src/class-wpml-beaver-builder-register-strings.php

```diff
@@ -20,4 +21,47 @@ protected function register_strings_for_modules( array $data_array, array $packa
 			}
 		}
 	}
+
+	/**
+	 * The modules are not in the order they appear on the page,
+	 * so we need to sort it before to register the strings.
+	 *
+	 * @param array $modules
+	 *
+	 * @return array
+	 */
+	private function sort_modules_before_string_registration( array $modules ) {
+		uasort( $modules, array( $this, 'sort_modules_by_position_only' ) );
+		return $this->sort_modules_by_parent_and_child( $modules );
+	}
+
+	/**
+	 * @param array        $all_modules
+	 * @param string|null  $parent_hash
+	 * @param array        $sorted_modules
+	 *
+	 * @return array
+	 */
+	private function sort_modules_by_parent_and_child( array $all_modules, $parent_hash = null, array $sorted_modules = array() ){
+		foreach ( $all_modules as $hash => $module ) {
+
+			if ( $module->parent === $parent_hash ) {
+				$sorted_modules[ $hash ] = $module;
+				unset( $all_modules[ $hash ] );
+				$sorted_modules = $this->sort_modules_by_parent_and_child( $all_modules, $module->node, $sorted_modules );
+			}
+		}
+
+		return $sorted_modules;
+	}
+
+	/**
+	 * @param stdClass $a
+	 * @param stdClass $b
+	 *
+	 * @return int
+	 */
+	private function sort_modules_by_position_only( stdClass $a, stdClass $b ) {
+		return ( (int) $a->position < (int) $b->position ) ? -1 : 1;
```

Comment:
@strategio You can simplify it with `return (int) $a->position - (int) $b->position`

Reply:
See here: https://gist.github.com/andreasciamanna/4031fa35830af267d7986494cd3d210e

Reply:
@andreasciamanna, as I mentionned in the snippet, this does not behave exactly the same. Anyway I tested your suggestion and the related test is failing, so I prefer to keep it as is.